great_expectations/metrics/metric.py

```diff
@@ -30,6 +34,8 @@ def __new__(cls, name, bases, attrs):
             or not any(issubclass(base_type, Domain) for base_type in bases)
         ):
             raise MixinTypeError(name, "Domain")
+        if name != "Metric" and not attrs.get("name"):
+            raise MissingAttributeError(name, "name")
```

Comment:
is this necessary? i would expect pydantic to handle this for us.

Reply:
i see that it doesn't. However, I think we can still remove this. We don't expect users to encounter this error, just developers who are working on the library.

Reply:
This is an open source library and I would hope one day we have external contributors defining metrics. Even if only internal developers encounter it, I don't see any harm in this.

Reply:
Ok you talked me into removing this. I think we would prefer in the future to not have to define this. It's just required because of the legacy system. Not raising an error in `Metric` makes it clear that only use it because we have to.